File: src/main/java/wraith/fabricaeexnihilo/util/PlayerHelper.java
package wraith.fabricaeexnihilo.util;

import net.minecraft.entity.effect.StatusEffectInstance;
import net.minecraft.entity.player.PlayerEntity;

public final class PlayerHelper {

    private PlayerHelper() {
    }

    public static void applyPotion(PlayerEntity player, StatusEffectInstance effect) {
        var playerEffect = player.getStatusEffect(effect.getEffectType());
        if ((playerEffect == null ? 0 : playerEffect.getDuration()) <= effect.getDuration()) {
            player.addStatusEffect(effect);
        }
    }
}

File: src/main/kotlin/exnihilofabrico/util/PlayerHelper.kt
package exnihilofabrico.util

import net.minecraft.entity.effect.StatusEffectInstance
import net.minecraft.entity.player.PlayerEntity

object PlayerHelper {
    fun applyPotion(player: PlayerEntity, effect: StatusEffectInstance) {
        if(player.getStatusEffect(effect.effectType)?.duration ?: 0 <= effect.duration)
            player.addStatusEffect(effect)
    }
}